src/DocumentDB.ChangeFeedProcessor/PartitionManagement/LeaseManagerBuilder.cs

```diff
@@ -63,19 +63,21 @@ public async Task<ILeaseManager> BuildAsync()
 
             this.leaseDocumentClient = this.leaseDocumentClient ?? this.leaseCollectionLocation.CreateDocumentClient();
 
-            if (this.leaseStoreCollectionLink == null)
+            if (this.leaseCollectionMetadata == null)
             {
-                DocumentCollection documentCollection = await this.leaseDocumentClient.GetDocumentCollectionAsync(this.leaseCollectionLocation).ConfigureAwait(false);
-                this.leaseStoreCollectionLink = documentCollection.SelfLink;
+                DocumentCollection leaseCollection = await this.leaseDocumentClient.GetDocumentCollectionAsync(this.leaseCollectionLocation).ConfigureAwait(false);
+                this.leaseCollectionMetadata = new CollectionMetadata(
+                    leaseCollection.SelfLink,
+                    leaseCollection.PartitionKey != null && leaseCollection.PartitionKey.Paths != null && leaseCollection.PartitionKey.Paths.Count != 0);
```

Comment:
Just one nit: Do we want to validate here that the PK is /id? And throw or do something otherwise?

Reply:
Good point. Currently this code path cannot be hit because we always provide the metadata for lease mgr builder, but I will extract and use same method in both cases.